plugins/WindowManager/InputMethodManager.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2019 UBports Foundation.
+ * Author(s) Marius Gripsgard <marius@ubports.com>
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3, as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+ * SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
```

Comment:
Hmm, none of the headers in this directory are using `#pragma once`. For consistency, please use a normal header guard define.

```suggestion
#ifndef INPUTMETHODMANAGER_H
#define INPUTMETHODMANAGER_H
```

Reply:
I disagree, we should use newer c++ when possible, using pragma once is faster, cleaner and safer. I have been using pragma once everywhere else (on other parts).

Reply:
Hmm, IMHO if we're gonna use pragma, we should migrate at the repository level, and as a separated PR.

@UniversalSuperBox, could you please comment?

Reply:
It's supported by both gcc (since 3.4) and clang (since somewhere around 2008), the only compilers I can see us ever being worried about. If both serve roughly the same purpose, `#pragma once` seems like the more clear define guard.

Reply:
Wait a minute, from https://gcc.gnu.org/onlinedocs/gcc-2.95.3/cpp_1.html#SEC8

>There is also an explicit directive to tell the preprocessor that it need not include a file more than once. This is called `#pragma once`, and was used in addition to the `#ifndef` conditional around the contents of the header file. **`#pragma once` is now obsolete and should not be used at all.**

(Emphasis mine)

Reply:
My goodness, what a complex issue. [Microsoft recommends its use](https://docs.microsoft.com/en-us/cpp/preprocessor/once?view=vs-2019), [GCC](https://gcc.gnu.org/onlinedocs/gcc-2.95.3/cpp_1.html#SEC8) and [ARM](http://www.keil.com/support/man/docs/armcc/armcc_chr1359124989221.htm) do not, Intel appears to have removed it from the newer versions of their compiler, the community loves it...

Ultimately, it does not matter which is used. Please create a PR to replace this `#pragma once` if you'd rather change it.

Reply:
The one from GCC is version 2.95 released decades ago, the reason why that was a thing is there was a bug [1] that has been fixed. New docs [2] says nothing about it being obsolete, it only says it's a less portable version and that is true but we use compiler flags that is less portable too so for us it does not matter (as it's way more portable then some flags we use). What other compiler do we expect to use that does not support pragma once? See https://en.wikipedia.org/wiki/Pragma_once#Portability

Also this sums it up pretty nicely https://www.reddit.com/r/cpp/comments/4cjjwe/come_on_guys_put_pragma_once_in_the_standard/

[1] https://gcc.gnu.org/bugzilla/show_bug.cgi?id=11569
[2] https://gcc.gnu.org/onlinedocs/gcc-9.2.0/cpp/Pragmas.html#Pragmas